Add a generic implementation for Daily queries - a loosely typed implementation over IQuery instead of a strongly typed implementation over QueryBase

eladBenedict opened the pull request.

eladBenedict committed:
Initial implementation

eladBenedict committed:
Merge branch 'master' into generateQueries

eladBenedict committed:
Update OS column name

eladBenedict committed:
Add generated Query per table

eladBenedict committed:
1. Remove storage-api module and incorporate into storage-access module

2. Add the newly generated table accessor list (previously in storage-api)

eladBenedict committed:
1. Decouple table column name from Dimensions

2. Generate table accessor list
3. Add additional basic column information to IColumnDefinition
4. Add missing "metric" field in tables metadata + update operating_system and partner column names according to Orly's script
5. Query generation (per table)

eladBenedict committed:
Some cosmetics and comments

eladBenedict committed:
Add a generic implementation for Daily queries - a loosely typed implementation over IQuery instead of a strongly typed implementation over QueryBase

eladBenedict committed:
initial implementation of computed metric/dimension query composition

eladBenedict committed:
Implement a computed metric

eladBenedict committed:
composite query support

eladBenedict committed:
Refactor

eladBenedict committed:
Additional refactoring

eladBenedict committed:
(Yet additional) refactoring

eladBenedict committed:
Merge pull request #18 from kaltura/queryComposition

Query composition

eladBenedict committed:
Implement EstimatedMinutesWatchedQuery

eladBenedict committed:
Separate externally visible metrics from internally available metrics

eladBenedict committed:
Extract a base class for all computed metrics and implement estimated minutes viewed,  average time viewed and average view drop-off

eladBenedict committed:
if there were no plays the drop off average should be 0

eladBenedict committed:
same for average time viewed

ofirski commented on kanalony-storage-logic/src/main/scala/kanalony/storage/logic/QueryLocator.scala:
```diff
@@ -38,19 +39,40 @@ object QueryLocator {
-          }
-        })
+      val relevantMetricsSupportedByQuery = q._1.supportedMetrics.intersect(remainingMetricsToCover).toList
+      if (!relevantMetricsSupportedByQuery.isEmpty)
```
Can be positive:
`if (relevantMetricsSupportedByQuery.nonEmpty)`

ofirski commented on kanalony-storage-access/build.sbt:
```diff
@@ -9,7 +9,8 @@ lazy val `kanalony-storage-access` = (project in file(".")).
       "com.websudos"            %% "phantom-dsl"                % "1.22.0",
-      "com.websudos"            %% "phantom-testkit"            % "1.12.2"
+      "com.websudos"            %% "phantom-testkit"            % "1.12.2",
+      "com.typesafe"            % "config"                      % "1.3.0"
```
Is there a reason not to use the ConfigurationManager on the core?

eladBenedict committed:
Merge branch 'master' into addDailyQuery

Conflicts:
	kanalony-storage-access/src/main/scala/kanalony/storage/generated/minutely_ua_prtn_entry_app_clst_playback_contextTableAccessor.scala
	kanalony-storage-access/src/main/scala/kanalony/storage/generator/TablesMetadata.scala
	kanalony-storage-logic/src/main/scala/kanalony/storage/logic/queries/DailyUserActivityPrtnEntryQuery.scala

eladBenedict committed:
Merge pull request #19 from kaltura/implementAdditionalComputedMetrics

Implement additional computed metrics

eladBenedict committed:
Merge remote-tracking branch 'origin/addDailyQuery' into addDailyQuery

eladBenedict committed:
1. Added support for additional inferred columns (e.g. month/day)

2. Updated table metadata with new tables and updated inferred columns
3. Generated table accessor and table row types
4. Generated corresponding queries and queryList

eladBenedict committed:
Merge pull request #20 from kaltura/updateTablesAndQueries

Update tables and queries
